content/2. Ondersteunende-informatie/3. Processen/3. Ontwerpproces/1. Opstellen ontwerp/Schematechnieken/1. Klassendiagram.md

```diff
@@ -0,0 +1,163 @@
+---
+title:
+  - Het klassendiagram
+tags:
+  - UML
+  - Klassendiagram
+difficulty: 2
+taxonomie:
+  - oo-15.1.Klassendiagram.OI
+  - oo-15.2.Klassendiagram.OI
+draft: true
+---
+
+## Wat is een klassendiagram
+
+Het klassendiagram is een ontwerptechniek voor het modelleren van de zogenaamde `structuur` van een softwaresysteem. Het bevat informatie over de `eigenschappen`(attributen) en het `gedrag` (methoden) van de klassen in het systeem. Ook bevat het diagram informatie over de `samenhang` (relaties) tussen de diverse klassen. 
+
+Onderstaand diagram is een voorbeeld van een klassendiagram. Hierin zie je de structuur van een eenvoudig hotel softwaresysteem. Dit voorbeeld is netjes uitwerkt maar kan zijn ontstaan terwijl je met je collega's bij een whiteboard stond.
+
+```mermaid
+classDiagram
+
+class Hotel {
+    +String naam
+    +String locatie
+    +geefKamers(bedrag)
+}
+
+
+class Kamer {
+	+String kamerNummer
+	+int aantalBedden
+	+decimal prijsPerNacht
+}
+
+Hotel --> Kamer
+```
+*Diagram 1: een eenvoudig voorbeeld van een klassendiagram*
+
+Onderstaande foto is het resultaat van een `brainstorm` op een whiteboard. Een nuttige toepassing als je nog in de ideeën fase bent.
+![[OI_Klassendiagram_1.jpg]]
+*Figuur 1: Een voorbeeld van een uitwerking op het whiteboard*
+## Hoe zit het klassendiagram in elkaar
+
+Klassendiagrammen kun je tegenkomen in een `functioneel ontwerp` en in een `technisch ontwerp`. In een functioneel ontwerp komt geen techniek voor. Je opdrachtgever of eindgebruiker zal een klassendiagram dat in een functioneel ontwerp is opgenomen kunnen begrijpen. Het beschrijft het domein waarvoor we een applicatie gaan maken. We noemen een dergelijk klassendiagram dan ook een `domeinmodel`.
+
+> [!info] Domeinmodel versus technisch klassendiagram
+>  - Door het opstellen van een domeinmodel krijg je inzicht in het domein(het probleemgebied) waarvoor je het systeem gaat ontwikkelen. (Bijvoorbeeld een hotelketen) 
+> - Door het opstellen van een technisch klassendiagram krijg je inzicht in de structuur van je applicatie. Dit gaat zowel over het domein als over de technische werking. (Bijvoorbeeld hoe de `User Interface` is gestructureerd of dat je `Events` gaat gebruiken)
+
+`Diagram 1` was een voorbeeld van een technisch klassendiagram. Hieronder vind je een voorbeeld van het bijbehorende domeinmodel.
+```mermaid
+classDiagram
+
+class Hotel {
+    naam
+    locatie
+}
+
+
+class Kamer {
+	kamernummer
+	aantal bedden
+	prijs per nacht
+}
+
+Hotel "1" -- "*" Kamer : heeft
+```
+*Diagram 2: een eenvoudig voorbeeld van een klassendiagram. In dit geval een domeinmodel. *
```

Comment:
Hier moet de nummering van het diagram 1.1 zijn, want het zit in bestand nummer 1 en is het eerste diagram
```suggestion
*Diagram 1.2: een eenvoudig voorbeeld van een klassendiagram. In dit geval een domeinmodel. *
```